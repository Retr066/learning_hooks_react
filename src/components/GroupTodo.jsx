import React from "react";
import Todo from "./Todo";

export default function GroupTodo({ todos, dispatch }) {
  const handleDelete = (todoId) => {
    const action = {
      type: "delete",
      payload: todoId,
    };
    dispatch(action);
  };
  const handleToogleCompleted = (todoId) => {
    const action = {
      type: "toogle",
      payload: todoId,
    };
    dispatch(action);
  };
  return (
    <>
      <ol className="list-group list-group-numbered">
        {todos.length >= 1 ? (
          todos.map((todo) => (
            <Todo
              key={todo.id}
              handleToogleCompleted={handleToogleCompleted}
              handleDelete={handleDelete}
              todo={todo}
            />
          ))
        ) : (
          <div className="alert alert-info" role="alert">
            No more homework
          </div>
        )}
      </ol>
    </>
  );
}
