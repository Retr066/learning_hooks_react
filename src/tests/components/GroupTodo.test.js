import { demoTodos } from "../fixtures/demoTodos";
import "@testing-library/jest-dom";
import GroupTodo from "../../components/GroupTodo";
import { shallow } from "enzyme";
describe("Pruebas para el componente GroupTodo", () => {
  const dispatch = jest.fn();

  let wrapper = shallow(<GroupTodo todos={demoTodos} dispatch={dispatch} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<GroupTodo todos={demoTodos} dispatch={dispatch} />);
  });

  test("debe renderizar correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("debe tener los elementos respectivos del array (2)", () => {
    expect(wrapper.find("Todo").length).toBe(demoTodos.length);
    expect(wrapper.find("Todo").at(0).prop("handleDelete")).toEqual(
      expect.any(Function)
    );
  });
});
