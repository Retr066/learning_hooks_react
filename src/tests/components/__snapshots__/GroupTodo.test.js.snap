// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente GroupTodo debe renderizar correctamente 1`] = `
<Fragment>
  <ol
    className="list-group list-group-numbered"
  >
    <Todo
      handleDelete={[Function]}
      handleToogleCompleted={[Function]}
      key="1"
      todo={
        Object {
          "completed": true,
          "desc": "Study react hooks (useReducer)",
          "id": 1,
          "title": "Learnig react",
        }
      }
    />
    <Todo
      handleDelete={[Function]}
      handleToogleCompleted={[Function]}
      key="2"
      todo={
        Object {
          "completed": false,
          "desc": "Study react hooks (test)",
          "id": 2,
          "title": "Learnig test in react js",
        }
      }
    />
  </ol>
</Fragment>
`;
